app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,44 @@
+class RestaurantsController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :set_restaurant, only: [:show, :edit, :update, :destroy]
+
+  def index
+    render json: {restaurants: ActiveModel::SerializableResource.new(Restaurant.sorted_by_name, each_serializer: RestaurantSerializer) }
+  end
+
+  def show
+    render json: { restaurant: RestaurantSerializer.new(@restaurant) }
+  end
+
+  def new
+    render json: { restaurant: RestaurantSerializer.new(Restaurant.new) }
+  end
+
+  def create
+    render json: { restaurant: RestaurantSerializer.new(Restaurant.create(restaurants_params)) }
+  end
+
+  def edit
+    render json: { restaurant: RestaurantSerializer.new(@restaurant) }
+  end
+
+  def update
+    @restaurant.update_attributes(restaurants_params) if @restaurant
+  end
+
+  def destroy
+    if @restaurant
+      @restaurant.reviews.each{|review| review.destroy }
```

Comment:
That's very nice, but what if you destroy a restaurant in any other place in your code (e.g services)? I recommend reading about something called "dependant: destroy", on the model level